Initial commit for Day-2 cluster profile variable 4.6.0 content; brea… (#5593)
spectrocloud/librarium: docs/docs-content/profiles/cluster-profiles/create-cluster-profiles/define-profile-variables/delete-cluster-profile-variables.md

1. Log in to [Palette](https://spectrocloud.com).

2. Navigate to the left **Main Menu** and select **Clusters**.

3. Select the host cluster you want to access.

4. From the cluster overview page, navigate to the middle column containing cluster details and locate the **Kubernetes
   Config File** row.

5. Click the **kubeconfig** link to download the file.

    ![Arrow pointing to the kubeconfig file.](/clusters_cluster-management_palette-webctl_cluster-details-overview.webp)

6. Open a terminal window and set the `KUBECONFIG` environment variable to the file path of your **kubeconfig** file. Below is an example.

    ```shell
    export KUBECONFIG=~/Downloads/dev-cluster.kubeconfig
    ```

7. Issue the appropriate `kubectl` command to verify that your parameter was populated correctly. 

   The following example searches the `cost-analyzer-cost-analyzer` deployment file in the `kubecost` namespace for the text `imagePullPolicy`, verifying that the `imagePullPolicy` for the Kubecost frontend was updated from `IfNotPresent` to `Always`.

    ```shell

>     kubectl get deployment cost-analyzer-cost-analyzer --namespace kubecost --output yaml | grep -C 2 "imagePullPolicy"

Author: no long form available

   profile variables. We recommend
   [creating a new version of your cluster profile](../../../cluster-profiles/modify-cluster-profiles/version-cluster-profile.md)
   before proceeding.
3. In the upper-right corner, click **Variables**.
4. Any variables present in a profile layer are indicated by a darkened pill and the text **[Count] layers**. Hover over
   the pill to review the profile layers that the variable is used in. You must remove the profile variable definition
   from all layers before you can delete it.

5. Close the **Profile variables** pane and open the profile layer containing the variable you want to remove.

6. Navigate to the pack's YAML configuration. Locate the profile variable, and remove
   `"{{.spectro.var.variable_name}}"`. Enter the appropriate value in its place.

   In the below example, we removed `"{{.spectro.var.kubecost-frontend-image-pull-policy}}"` and set
   `kubecostFrontend.imagePullPolicy` to `Always`.

   ![Removing a cluster profile variable from YAML configuration](/profiles_cluster-profiles_create-cluster-profiles_define-profile-variables_delete-cluster-profile-variables_remove-from-YAML.webp)

> 7. Continue removing all references to the profile variable from the layer. When you are finished, select **Confirm

Reviewer: ```suggestion
7. Remove all references to the profile variable from the layer. When you are finished, select **Confirm
```

   Updates**.

8. Repeat steps five through seven as necessary until the profile variable is removed from all layers and the pill
   beside the variable changes to **unused**.

9. Close the **Profile variables** pane, and on the main cluster profile window, select **Save Changes**.

10. Once your changes have been saved, in the upper-right corner, click **Variables**.
11. Select the **three-dot Menu** beside the **unused** cluster profile variable. **Delete** the variable.

> ### Validation

Reviewer: ```suggestion
### Validate
```

1.  Log in to [Palette](https://console.spectrocloud.com).
2.  From the left **Main Menu**, select **Clusters**. Choose the cluster from which you are removing the cluster profile
    variable.

3.  Navigate to the **Profile** tab of the cluster, where you can quickly view how many profile variables are being used
    in the current version of your cluster profile.

4.  If you need to update a cluster profile version to one without the profile variable, expand the semantic version
    **drop-down Menu** and choose the appropriate version.

5.  If you need to replace or remove a profile, select the **three-dot Menu** beside the profile, and choose the
    applicable action.

6.  Select **Review & Save**. The **Changes Summary** dialog appears. Select **Review changes in Editor**.

7.  Use the **Profile variable changes** tab to view the profiles that were changed. Expand each profile to compare the
    **Running configuration** and **New configuration**, making any necessary changes to your cluster profile variables.
    Updated profile variable values must match the schema defined in the cluster profile variable.

>     Each profile must be **Reviewed**, indicated by a green check mark, before you can apply your changes. In the

Reviewer: ```suggestion
    Each profile must have the **Reviewed** status, indicated by a green check mark, before you can apply your changes. In the
```
maybe??

profile.

:::warning

We do not recommend updating a currently deployed cluster profile version to remove profile variables. Instead, we
recommend creating a new profile version,
[removing them from the profile](#delete-profile-variables-from-a-cluster-profile), and then upgrading active clusters
to the new version. For information on versioning profiles, check out
[Version a Cluster Profile](../../../cluster-profiles/modify-cluster-profiles/version-cluster-profile.md).

:::

1. Log in to [Palette](https://console.spectrocloud.com).
2. From the left **Main Menu**, select **Clusters**. Choose the cluster from which you are removing the cluster profile
   variable.

3. Navigate to the **Profile** tab of the cluster.

4. Navigate to the pack's YAML configuration. Locate the profile variable, and remove
   `"{{.spectro.var.variable_name}}"`. Enter the appropriate value in its place.

5. Repeat the previous step until you have removed all applicable profile variables.

6. **Save** your changes.

7. Your cluster begins the update process. Navigate to the cluster **Events** tab to monitor your cluster's progress.

</TabItem>

</Tabs>

> ### Validation

Reviewer: ```suggestion
### Validate
```

> Once your cluster is finished updating, verify that any parameters previously containing cluster profile variables were
> populated with the expected values.
> 
> The following validation process uses the [kubectl CLI](https://kubernetes.io/docs/reference/kubectl/) and the cluster's
> [kubeconfig](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) file. For more
> information on how to use `kubectl` and **kubeconfig** with your Palette clusters, refer to the Spectro Cloud
> [Kubectl](../../../../clusters/cluster-management/palette-webctl.md) and
> [Kubeconfig](../../../../clusters/cluster-management/kubeconfig.md) guides.

Reviewer: ```suggestion
Use the following steps to verify that any parameters previously containing cluster profile variables were
populated with the expected values.
```
Same feedback about replacing steps 4-6 with the kubectl guide

1. Log in to [Palette](https://spectrocloud.com).

2. Navigate to the left **Main Menu** and select **Clusters**.

3. Select the host cluster you want to access.

4. From the cluster overview page, navigate to the middle column containing cluster details and locate the **Kubernetes
   Config File** row.

5. Click the **kubeconfig** link to download the file.

   ![Arrow pointing to the kubeconfig file.](/clusters_cluster-management_palette-webctl_cluster-details-overview.webp)

6. Open a terminal window and set the `KUBECONFIG` environment variable to the file path of your **kubeconfig** file.
   Below is an example.

   ```shell
   export KUBECONFIG=~/Downloads/dev-cluster.kubeconfig
   ```

7. Issue the appropriate `kubectl` command to verify that your parameter was populated correctly.

   The following example searches the `cost-analyzer-cost-analyzer` deployment file in the `kubecost` namespace for the
   text `imagePullPolicy`, verifying that the `imagePullPolicy` for the Kubecost frontend was updated from
   `IfNotPresent` to `Always`.

   ```shell

>    kubectl get deployment cost-analyzer-cost-analyzer --namespace kubecost --output yaml | grep -C 2 "imagePullPolicy"

Author: No long form available
